### PR TITLE
[Feature] Support custom persistent_workers

### PR DESCRIPTION
**Pull request by hhaAndroid**
## Motivation

Support custom `persistent_workers`

Note: The persistent_workers setting is invalid when the pytorch version is lower than 1.7.

### CODE REVIEW

File: mmdet/datasets/builder.py [hunk 1]
```diff
@@ -163,14 +173,18 @@ def build_dataloader(dataset,
         worker_init_fn, num_workers=num_workers, rank=rank,
         seed=seed) if seed is not None else None
 
+    if (TORCH_VERSION != 'parrots'
+            and digit_version(TORCH_VERSION) >= digit_version('1.7.0')):
+        kwargs['persistent_workers'] = persistent_workers
```
Comment (RangiLyu) [inline mmdet/datasets/builder.py:178]: else:
      warning.warn(...)

File: mmdet/datasets/builder.py [hunk 2]
```diff
@@ -106,7 +109,15 @@ def build_dataloader(dataset,
         dist (bool): Distributed training/test or not. Default: True.
         shuffle (bool): Whether to shuffle the data at every epoch.
             Default: True.
+        seed (int, Optional): Seed to be used. Default: None.
         runner_type (str): Type of runner. Default: `EpochBasedRunner`
+        pin_memory (bool): Whether to use pin_memory in DataLoader.
+            Default: False.
+        persistent_workers (bool): If True, the data loader will not shutdown
+            the worker processes after a dataset has been consumed once.
+            This allows to maintain the workers Dataset instances alive.
```
Comment (RangiLyu) [inline mmdet/datasets/builder.py:118]:
```suggestion
            This allows to maintain the workers `Dataset` instances alive.
```

File: mmdet/datasets/builder.py [hunk 3]
```diff
@@ -106,7 +109,15 @@ def build_dataloader(dataset,
...
+            The argument also has effect in PyTorch>=1.7.0.
```
Comment (RangiLyu) [inline mmdet/datasets/builder.py:119]:
```suggestion
            This argument is only valid when PyTorch>=1.7.0.
```

**Review by RangiLyu — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

File: mmdet/datasets/builder.py [hunk 4]
```diff
@@ -163,14 +173,21 @@ def build_dataloader(dataset,
...
+
     data_loader = DataLoader(
         dataset,
         batch_size=batch_size,
         sampler=sampler,
         num_workers=num_workers,
         batch_sampler=batch_sampler,
         collate_fn=partial(collate, samples_per_gpu=samples_per_gpu),
-        pin_memory=False,
+        pin_memory=pin_memory,
```
Comment (ZwwWayne) [inline mmdet/datasets/builder.py:190]: For now, we actually do not support setting pin_memory through the config, should we allow that?

Comment (hhaAndroid) [reply mmdet/datasets/builder.py:190]: Ok. User is not allowed to set.

**Review by ZwwWayne — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**